applications/StructuralMechanicsApplication/custom_elements/shell_thick_element_3D3N.hpp

```diff
@@ -191,6 +191,13 @@ namespace Kratos
 
 		///@}
 
+        ///@name Member Variables
+		///@{
+
+		CoordinateTransformationBasePointerType mpCoordinateTransformation; /*!< The Coordinate Transformation */
```

Comment:
do you need it to be a pointer?

Reply:
In the context of our sensitivity analysis stuff in does not matter if it is a pointer or not. This question should be answered by the people who developed the shell elements. Maybe @philbucher knows a reason.

Reply:
I do :)

The shells handle the Kinematics through the class `CoordinateTransformation`
At the construction of the Shell-Element this pointer gets either filled with a linear or a corotational Class (which derives from then linear one) => the pointer is needed for the polymorphic behavior

@RiccardoRossi in another branch I am creating a base-class for the T3 and Q4 CoordinateTransformation then this pointer will be moved to the BaseClass